src/mapml/handlers/ContextMenu.js

```diff
@@ -646,17 +649,21 @@ export var ContextMenu = L.Handler.extend({
     copyEl.setAttribute("aria-expanded","true");
     menu.removeAttribute('hidden');
 
-    if (click.containerPoint.x + 160 + 80 > mapSize.x) {
+    const menuWidth = this._container.offsetWidth,
+          menuHeight = this._container.offsetHeight,
+          submenuWidth = menu.offsetWidth;
+    if (click.containerPoint.x + menuWidth + submenuWidth > mapSize.x) {
       menu.style.left = 'auto';
-      menu.style.right = 160 + 'px';
+      menu.style.right = menuWidth + 'px';
     } else {
-      menu.style.left = 160 + 'px';
+      menu.style.left = menuWidth + 'px';
       menu.style.right = 'auto';
     }
 
-    if (click.containerPoint.y + 160 > mapSize.y) {
+    if (click.containerPoint.y + menuHeight + 73 > mapSize.y) {
```

Comment:
I'm only left wondering what the 73 is for.  You replaced other 'magic numbers' with variables, which makes the code much more understandable.  Could we assign 73 to a variable and use that?

Reply:
> I'm only left wondering what the 73 is for. 

it is the height difference between the main context menu and the submenu.

> Could we assign 73 to a variable and use that?

Yes, it will make more sense to change it to a variable! I have pushed a new commit and I also add a comment for it.